streaming/base/dataset.py

```diff
@@ -280,13 +278,14 @@ class StreamingDataset(Array, IterableDataset):
             how many samples to pick from the same shard at a time (``1`` for evenly balanced
             across shards, ``1000`` to pick 1000 samples from the same shard at a time, etc).
             Defaults to ``1``.
-        partition_algo (str): Which partitioning algorithm to use. Defaults to ``orig``.
+        partition_algo (str): Which partitioning algorithm to use. Defaults to ``relaxed``.
         num_canonical_nodes (int, optional): Canonical number of nodes for shuffling with
             resumption. The sample space is divided evenly according to the number of canonical
             nodes. The higher the value, the more independent non-overlapping paths the
             StreamingDataset replicas take through the shards per model replica (increasing data
-            source diversity). Defaults to ``None``, which is interpreted as 64 times the number
-            of nodes of the initial run.
+            source diversity). If ``None``, this is interpreted as 64 times the number of physical
```

Comment:
Any reason why `64` for some shuffle algo and `1` for others?

Reply:
The `py1s` and `py2s` shuffles are completely dependent on high `num_canonical_nodes` to get a good shuffle, so having `ncn=64*physical_nodes` for those two is better. The other shuffle algos are shuffle-block-based, and so we are setting `num_canonical_nodes` to equal `physical_nodes` and then varying the `shuffle_block_size` to get good shuffle quality.